metricbeat/mb/registry.go

```diff
@@ -300,11 +300,33 @@ func (r *Register) Modules() []string {
 	r.lock.RLock()
 	defer r.lock.RUnlock()
 
+	var dups = map[string]bool{}
+
+	// For the sake of compatibility, grab modules the old way as well, right from the modules map
 	modules := make([]string, 0, len(r.modules))
 	for module := range r.modules {
+		dups[module] = true
 		modules = append(modules, module)
 	}
 
+	// List also modules from secondary sources
+	if source := r.secondarySource; source != nil {
+		sourceModules, err := source.Modules()
+		if err != nil {
+			logp.Error(errors.Wrap(err, "failed to get modules from secondary source"))
```

Comment:
I'm pretty sure this isn't what you want. `Error` a field type that can be used with structured logging to optionally specify type information.

I think you want either `logp.L().Error(errors.Wrap(...))` or `logp.L().Errorw("Failure getting module list from secondary source", "error", err)` or `logp.L().Errorf("Kaboom! error: %v", err)`.

Reply:
Bah, that code was all over `registry.go` so I just copied it. Thinking I should make a separate PR to fix it that can be backported.

Reply:
Not sure why I ended-up doing this, I was probably looking for something like `logp.Err` :grimacing: